freeipa/src/main/java/com/sequenceiq/freeipa/service/freeipa/user/UmsUsersStateProvider.java

```diff
@@ -84,11 +92,38 @@
             return envUsersStateMap;
         } catch (RuntimeException e) {
             throw new UmsOperationException(String.format("Error during UMS operation: '%s'", e.getLocalizedMessage()), e);
-        } finally {
-            requestIdThreadLocal.remove();
         }
     }
 
+    private WorkloadCredential getCredentials(String actorCrn, String userCrn, Optional<String> requestId) {
+        GetActorWorkloadCredentialsResponse response = grpcUmsClient.getActorWorkloadCredentials(actorCrn, userCrn, requestId);
+        String hashedPassword = response.getPasswordHash();
+        List<ActorKerberosKey> keys = response.getKerberosKeysList();
+        return new WorkloadCredential(hashedPassword, keys);
+    }
+
+    private void populateWorkloadCredentialMap(String actorCrn, Map<String, WorkloadCredential> usersWorkloadCredentialMap, List<User> users, List<MachineUser> machineUsers, Optional<String> requestId) {
```

Comment:
This method looks redundant with your changes to `handleUser`. You should remove it.

Reply:
Removed. Thanks.